hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -572,4 +574,18 @@ public ReadCommittedScope getReadCommitted(JobContext job, AbsoluteTableIdentifi
   public void setReadCommittedScope(ReadCommittedScope readCommittedScope) {
     this.readCommittedScope = readCommittedScope;
   }
+
+  public String getSegmentIdFromFilePath(String filePath) {
```

Comment:
Already `CarbonTablePath#getSegmentIdFromPath` method exist. check if you can use it.

Reply:
_CarbonTablePath#getSegmentIdFromPath_ returns segmentId present in path. For eg: Fact/Part0/segment_**0**. But in this case we need segment id present in carbondata file name. So above method cannot be used.

Reply:
The caller of this method, inputSplit itself has getSegmentId() method. I think you can use it instead of this. Please avoid writing new code and try to reuse the code.

Reply:
getSegmentId() also return segmentId from LoadMetaDataDetails of segment which is not null. Only null for segmentId is present in name of carbondata file of SDK segment and there is no already existing method to get that.

Reply:
ok